src/super_gradients/training/utils/callbacks/callbacks.py

```diff
@@ -1208,11 +1210,40 @@ def on_validation_loader_end(self, context: PhaseContext) -> None:
             self._reset()
 
     def _gather_extreme_batch_images_and_log(self, context, loader_name: str):
-        images_to_save = self.process_extreme_batch()
-        images_to_save = maybe_all_gather_np_images(images_to_save)
-        if self.max_images > 0:
-            images_to_save = images_to_save[: self.max_images]
+        from pytorch_toolbelt.utils.distributed import all_gather
```

Comment:
pytorch_toolbelt?

Reply:
That was leaked by accident from another branch ^_^